controllers/imageclusterinstall_controller.go

```diff
@@ -711,8 +718,19 @@ func (r *ImageClusterInstallReconciler) removeBMHDataImage(ctx context.Context,
 
 	if bmh != nil {
```

Comment:
```suggestion
	if bmh != nil {
```
```suggestion
	if bmh == nil {
	return
	}
```

Reply:
Why? BMH shouldn't be nil if we want to change/remove some of its annotations